### PR TITLE
Update build ci image

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Update script!
New workflow:
1. always checkout `push-ci-image`
2. do your thing
3. push a commit with [build-ci-image]
4. check the build
5. push a commit with [push-ci-image]
6. close the pr and delete the branch. 
7. DONE

### CODE REVIEW

**Comment by ArthurZucker** [comment]
I need this to merge #30932

**Review by LysandreJik — APPROVED** [review]
Make sure Yih-Dar has seen the PR as well

File: .github/workflows/build-ci-docker-images.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
   build:
     runs-on: ubuntu-22.04
 
-    if: ${{ contains(github.event.head_commit.message, '[push-ci-image]') && '!cancelled()' }}
+    if: ${{ contains(github.event.head_commit.message, '[build-ci-image]') || contains(github.event.head_commit.message, '[push-ci-image]') && '!cancelled()' || github.event_name == 'schedule' }}
```
Comment (ydshieh) [inline .github/workflows/build-ci-docker-images.yml:25]: I guess `[build-ci-image]` is to build but not to push, right?

Comment (ydshieh) [reply .github/workflows/build-ci-docker-images.yml:25]: oh, I see you mentioned it in PR description.

**Comment by ydshieh** [comment]
Looks good to me. Just to comment that, by `delete the branch` you mean the remote branch I believe.

**Review by ydshieh — APPROVED** [review]
Thanks for the update. In terms of the building itself, it's OK.

But taking into account the CircleCI runs, I think we have more to do.

**Comment by ArthurZucker** [comment]
Yeah I'm pinning the corresponding timm version

**Comment by ArthurZucker** [comment]
Actually I should probably build to push a different image than latest, otherwise it still affects everyone else 😅

File: .circleci/config.yml [hunk 2]
```diff
@@ -8,6 +8,12 @@ parameters:
         type: boolean
         default: false
 
+on:
+    workflow_run:
+        workflows: [" Build pr ci-docker"]
+        types:
+        - completed
+
```
Comment (ArthurZucker) [inline .circleci/config.yml:16]:
```suggestion
```
since the workflow is from gh action, does not work

File: setup.py [hunk 3]
```diff
@@ -176,7 +176,7 @@
     "tensorflow-probability<2.16",
     "tf2onnx",
     "timeout-decorator",
-    "timm",
+    "timm<=0.9.16",
```
Comment (ArthurZucker) [inline setup.py:179]: @amyeroberts I ended up pinning this, feel free to unpin once you feel like it

**Pull request merged**